packages/panzer/disc-fe/src/Panzer_BasisValues2_impl.hpp

```diff
@@ -322,1434 +275,1494 @@ evaluateValues(const PHX::MDField<Scalar,Cell,IP,Dim> & cub_points,
+template <typename Scalar>
+void
+BasisValues2<Scalar>::
+setOrientations(const Teuchos::RCP<const OrientationsInterface> & orientations)
+{
+  TEUCHOS_TEST_FOR_EXCEPT_MSG(orientations == Teuchos::null,
+                              "BasisValues2::setOrientations : Orientations have already been set.");
+  orientations_ = orientations;
 
-  if(build_weighted){
-    const std::pair<int,int> cell_range(0,num_cells);
-    auto s_weighted_basis_vector = Kokkos::subview(weighted_basis_vector.get_view(),cell_range,Kokkos::ALL(),Kokkos::ALL(),Kokkos::ALL(),Kokkos::ALL(),Kokkos::ALL(),Kokkos::ALL());
-    auto s_weighted_measure = Kokkos::subview(weighted_measure.get_view(),cell_range,Kokkos::ALL());
-    auto s_basis_vector = Kokkos::subview(basis_vector.get_view(),cell_range,Kokkos::ALL(),Kokkos::ALL(),Kokkos::ALL());
-    fst::multiplyMeasure(s_weighted_basis_vector,s_weighted_measure,s_basis_vector);
-    if(compute_derivatives){
-      if(num_dim==2){
-        auto s_weighted_curl_basis_scalar = Kokkos::subview(weighted_curl_basis_scalar.get_view(),cell_range,Kokkos::ALL(),Kokkos::ALL());
-        auto s_curl_basis_scalar = Kokkos::subview(curl_basis_scalar.get_view(),cell_range,Kokkos::ALL(),Kokkos::ALL());
-        fst::multiplyMeasure(s_weighted_curl_basis_scalar,s_weighted_measure,s_curl_basis_scalar);
-      } else if(num_dim==3){
-        auto s_weighted_curl_basis_vector = Kokkos::subview(weighted_curl_basis_vector.get_view(),cell_range,Kokkos::ALL(),Kokkos::ALL(),Kokkos::ALL());
-        auto s_curl_basis_vector = Kokkos::subview(curl_basis_vector.get_view(),cell_range,Kokkos::ALL(),Kokkos::ALL(),Kokkos::ALL());
-        fst::multiplyMeasure(s_weighted_curl_basis_vector,s_weighted_measure,s_curl_basis_vector);
-      }
-    }
-  }
+  // Setting orientations means that we need to reset our arrays
+  resetArrays();
+}
 
+template <typename Scalar>
+void
+BasisValues2<Scalar>::
+resetArrays()
+{
+  // Turn off all evaluated fields (forces re-evaluation)
+  basis_ref_scalar_evaluated_ = false;
+  basis_scalar_evaluated_ = false;
+  basis_ref_vector_evaluated_ = false;
+  basis_vector_evaluated_ = false;
+  grad_basis_ref_evaluated_ = false;
+  grad_basis_evaluated_ = false;
+  curl_basis_ref_scalar_evaluated_ = false;
+  curl_basis_scalar_evaluated_ = false;
+  curl_basis_ref_vector_evaluated_ = false;
+  curl_basis_vector_evaluated_ = false;
+  div_basis_ref_evaluated_ = false;
+  div_basis_evaluated_ = false;
+  weighted_basis_scalar_evaluated_ = false;
+  weighted_basis_vector_evaluated_ = false;
+  weighted_grad_basis_evaluated_ = false;
+  weighted_curl_basis_scalar_evaluated_ = false;
+  weighted_curl_basis_vector_evaluated_ = false;
+  weighted_div_basis_evaluated_ = false;
+  basis_coordinates_ref_evaluated_ = false;
+  basis_coordinates_evaluated_ = false;
+
+  // TODO: Enable this feature - requires the old interface to go away
+  // De-allocate arrays if necessary
+//  if(not alloc_arrays){
+//    basis_ref_scalar = Array_BasisIP();
```

Comment:
@seamill - this is the correct way to delete the allocated memory.

Reply:
Ah, yes it will do that, but my understanding is that it resets the field itself (e.g. tag). Does that affect the BasisValues_Evaluator which registers the tag with the field manager. I guess it is only registering a pointer.